src/AddProjectsPage/ChosenTags.tsx

```diff
@@ -6,34 +6,32 @@ class ChosenTags extends React.Component<{
 }> {
   render() {
     var chosenTags;
-    if (!this.props.tags) {
+    let tags = this.props.tags;
+    if (!tags || tags.slice().length === 0) {
       return null;
     }
+    // make copy of tags
+    tags = tags.slice();
```

Comment:
We already sliced in line 10, can we use that result? Maybe we could also use a different variable name as well so you don't need to comment. e.g., if we look further down in the code we'll know that we are using a copy, but why are we using a copy here anyway?

We can separate this logic into its own function, we should try to limit `render` to only rendering.

Reply:
removed the move of tags, as you are right -- its unnecessary. Moved logic to its own function